src/Buildalyzer/AnalyzerManager.cs

```diff
@@ -63,12 +68,13 @@ public AnalyzerManager(string solutionFilePath, TextWriter logWriter, LoggerVerb
             {
                 solutionFilePath = ValidatePath(solutionFilePath, true);
                 SolutionDirectory = Path.GetDirectoryName(solutionFilePath);
-                GetProjectsInSolution(solutionFilePath);
+                GetProjectsInSolution(solutionFilePath, projects);
             }
         }
 
-        private void GetProjectsInSolution(string solutionFilePath)
+        private void GetProjectsInSolution(string solutionFilePath, string[] projects = null)
         {
+            projects = projects ?? new string[] { };

```

Comment:
Instead of allocating an empty array and then checking
```
if (projects.Length > 0 && !projects.Contains(project.ProjectName))
```
further down, I'm wondering if it would be better to skip the empty array and make the check
```
if(projects?.Contains(project.ProjectName) ?? false)
```

Thoughts?

Reply:
Much better! will amend

Reply:
Went with another variation in 8a3cf6b, let me know what you think :)